### PR TITLE
Added CONTRIBUTING.md based on original drafts

### PR DESCRIPTION
**Pull request by JeromSar**
This pull request incorporates contributing information based on the [original drafts](https://gist.github.com/SpaceManiac/a5f6d04c4f095c988fcc).

I attempted to carefully describe how a change should proceed. Included is in-depth detail describing what a contributor might expect.

Feedback is greatly appreciated.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+* No trailing whitespaces for code lines, comments or configuration files.
+* No CRLF line endings, only LF is allowed.
+  * For Windows-based machines, you can configure Git to do this for your by running `git config --global core.autocrlf true`.
+  * If you're running a Linux or Mac OSX, you should run `git config --global core.autocrlf input` instead.
+  * For more information about line feeds. See this [this article](http://adaptivepatchwork.com/2012/03/01/mind-the-end-of-your-line/).
+* Files must always end with a newline.
+* No 80 character line limit or 'weird' midstatement newlines.
+* Avoid nested code structures.
+* Use 4 spaces instead of tabs.
+* All files must end with a new line.
```
Comment (turt2live) [inline CONTRIBUTING.md]: Repeated

**Comment by turt2live** [comment]
No mention about documentation standards or other future requirements.

**Comment by JeromSar** [comment]
@turt2live What do you suggest for documentation standards?

**Comment by turt2live** [comment]
Standards set by Bukkit.

**Comment by turt2live** [comment]
Going to let @SpaceManiac deal with this one

**Comment by turt2live** [comment]
Ticket: #167

**Comment by SpaceManiac** [comment]
Manually merged in bbc396017a352c93b167e3a1bea5e940444e2667.

**Pull request closed without merging**